journal/src/main/java/io/camunda/zeebe/journal/file/SegmentWriter.java

```diff
@@ -230,26 +229,24 @@ private void reset(final long index, final long lastFlushedIndex) {
     } catch (final BufferUnderflowException e) {
       // Reached end of the segment
     } catch (final CorruptedJournalException e) {
-      handleChecksumMismatch(e, nextIndex, lastFlushedIndex, position);
+      if (detectCorruptionAsPartialWrite) {
+        resetPartiallyWrittenEntry(e, position);
+      } else {
+        throw e;
+      }
     } finally {
       buffer.reset();
     }
   }
 
-  private void handleChecksumMismatch(
-      final CorruptedJournalException e,
-      final long nextIndex,
-      final long lastFlushedIndex,
-      final int position) {
-    // entry wasn't acked (likely a partial write): it's safe to delete it
-    if (nextIndex > lastFlushedIndex) {
-      FrameUtil.markAsIgnored(buffer, position);
-      buffer.position(position);
-      buffer.mark();
-      return;
-    }
-
-    throw e;
+  private void resetPartiallyWrittenEntry(final CorruptedJournalException e, final int position) {
+    LOG.debug(
```

Comment:
:wrench: You can always pass the error as the last argument for all the logger functions, and it will log them properly. It seems a little strange to me to put the error message first, then the context. What does it look like on failure?

Reply:
See PR description. The log looks like this.
`Record's checksum (810675654) doesn't match checksum stored in metadata (2636955287). Found a corrupted or partially written entry at position 97. Considering it as a partially written entry and resetting the position.`

I do not want to log the whole stack trace here because it could be just a partial write. This is why I did not pass the error to the logger.